library/src/main/java/com/otaliastudios/zoom/internal/Controller.kt

```diff
@@ -0,0 +1,148 @@
+package com.otaliastudios.zoom.internal
+
+import android.view.MotionEvent
+import androidx.annotation.IntDef
+import com.otaliastudios.zoom.ZoomLogger
+
+/**
+ * Deals with touch input, holds the internal [state] integer,
+ * and applies special logic to touch inputs and state changes to
+ * prefer one gesture over the other, for example.
+ */
+internal class Controller(private val callback: Callback) {
```

Comment:
Imho the name `Controller` is way to generic. This controller could do literally anything.
It looks like it could be a `TouchInputController`?
Or maybe just an `InputController`?
Or simply a `ZoomEngineController`?